tests/unit/runtime/zero/test_zero_leaf_module.py

```diff
@@ -143,3 +188,69 @@ def test_set_no_match_class(self):
             raise AssertionError("Expected error that no module is set as a leaf module")
         except ValueError as e:
             pass
+
+
+class TestZ3LeafOptimization(DistributedTest):
+    world_size = 2
+    reuse_dist_env = True
+
+    def test_finegrained_optimization(self):
+        hidden_dim = 128
+        num_block = 16
+        stage3_coalesced_fetch_threshold = 12000
+        config_dict = {
+            "train_micro_batch_size_per_gpu": 1,
+            "steps_per_print": 1,
+            "optimizer": {
+                "type": "Adam",
+                "params": {
+                    "lr": 1e-6
+                }
+            },
+            "zero_optimization": {
+                "stage": 3,
+                "stage3_prefetch_bucket_size": hidden_dim**2,
+                "stage3_param_persistence_threshold": 0,
+                "stage3_max_reuse_distance": 0,
+                "stage3_coalesced_fetch_threshold": stage3_coalesced_fetch_threshold
+            }
+        }
+        if get_accelerator().is_fp16_supported():
```

Comment:
Please use  `preferred_dtype()` instead of `is_*_supported()` to ensure consistency with line 231.

Reply:
fixed